lib/FrontendTool/ReferenceDependencies.cpp

```diff
@@ -127,13 +127,33 @@ static std::string escape(DeclBaseName name) {
   return llvm::yaml::escape(name.userFacingName());
 }
 
-static void emitProvides(const SourceFile *SF, llvm::raw_ostream &out);
-static void emitDepends(const SourceFile *SF,
-                        const DependencyTracker &depTracker,
-                        llvm::raw_ostream &out);
-static void emitInterfaceHash(SourceFile *SF, llvm::raw_ostream &out);
+namespace {
+  class ReferenceDependenciesEmitter {
+    SourceFile * const SF;
+    const DependencyTracker &depTracker;
+    llvm::raw_ostream &out;
+    
+    ReferenceDependenciesEmitter(SourceFile *const SF, const DependencyTracker &depTracker, llvm::raw_ostream &out) :
+    SF(SF), depTracker(depTracker), out(out) {}
+    
+  public:
+    /// \return true on error
+    static bool emit(DiagnosticEngine &diags, SourceFile *const SF,
+                     const DependencyTracker &depTracker,
+                     StringRef outputPath);
+    static void emit(SourceFile *const SF, const DependencyTracker &depTracker, llvm::raw_ostream &out);
+    
+  private:
+    static std::unique_ptr<llvm::raw_fd_ostream> openFile(DiagnosticEngine &diags,
+                                                          StringRef OutputPath);
+    void emit();
+    void emitProvides();
+    void emitDepends();
+    void emitInterfaceHash();
+  };
+} // end anon namespace
 
-static std::unique_ptr<llvm::raw_fd_ostream>openFile(DiagnosticEngine &diags, StringRef outputPath) {
+std::unique_ptr<llvm::raw_fd_ostream>ReferenceDependenciesEmitter::openFile(DiagnosticEngine &diags, StringRef outputPath) {
```

Comment:
nit: I don't know if this is intentional but it looks like you're not leaving a space between the end of the unique_ptr<> type and the function name, which .. seems a bit off?